### PR TITLE
Fix TransformerTokenizer init for chatglm2 & 3

### LINKED ISSUE #735: [Bug] run chatglm3-6b report error
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.
- [X] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.

### Describe the bug

WARNING 07-26 14:06:20 interfaces.py:131] The model (<class 'sglang.srt.models.chatglm.ChatGLMForCausalLM'>) contains all LoRA-specific attributes, but does not set `supports_lora=True`.
Loading pt checkpoint shards:   0% Completed | 0/7 [00:00<?, ?it/s]
Loading pt checkpoint shards:  14% Completed | 1/7 [00:00<00:04,  1.28it/s]
Loading pt checkpoint shards:  29% Completed | 2/7 [00:01<00:04,  1.01it/s]
Loading pt checkpoint shards:  43% Completed | 3/7 [00:03<00:04,  1.05s/it]
Loading pt checkpoint shards:  57% Completed | 4/7 [00:04<00:03,  1.06s/it]
Loading pt checkpoint shards:  71% Completed | 5/7 [00:05<00:02,  1.09s/it]
Loading pt checkpoint shards:  86% Completed | 6/7 [00:06<00:01,  1.08s/it]
Loading pt checkpoint shards: 100% Completed | 7/7 [00:07<00:00,  1.09s/it]
Loading pt checkpoint shards: 100% Completed | 7/7 [00:07<00:00,  1.06s/it]

[gpu_id=0] Load weight end. type=ChatGLMForCausalLM, dtype=torch.float16, avail mem=11.55 GB
[gpu_id=0] Memory pool end. avail mem=1.83 GB
[gpu_id=0] Capture cuda graph begin. This can take up to several minutes.
[gpu_id=0] max_total_num_tokens=336793, max_prefill_tokens=16384, max_running_requests=21048, context_len=8192
Initialization failed. controller_init_state: Traceback (most recent call last):
  File "/home/admin/anaconda3/envs/sgl-env/lib/python3.10/site-packages/sglang/srt/managers/controller/manager_single.py", line 135, in start_controller_process
    controller = ControllerSingle(
  File "/home/admin/anaconda3/envs/sgl-env/lib/python3.10/site-packages/sglang/srt/managers/controller/manager_single.py", line 69, in __init__
    self.tp_server = ModelTpServer(
  File "/home/admin/anaconda3/envs/sgl-env/lib/python3.10/site-packages/sglang/srt/managers/controller/tp_worker.py", line 150, in __init__
    self.regex_fsm_cache = FSMCache(
  File "/home/admin/anaconda3/envs/sgl-env/lib/python3.10/site-packages/sglang/srt/constrained/fsm_cache.py", line 24, in __init__
    self.outlines_tokenizer = TransformerTokenizer(tokenizer)
  File "/home/admin/anaconda3/envs/sgl-env/lib/python3.10/site-packages/outlines/models/transformers.py", line 68, in __init__
    self.tokenizer.pad_token_id = self.tokenizer.eos_token_id
AttributeError: can't set attribute 'pad_token_id'

Initialization failed. detoken_init_state: init ok

### Reproduction

 command: python -m sglang.launch_server --model-path /home/admin/apps/models/chatglm3-6b/ --port 30000 --trust-remote-code

### Environment

```Shell
python 3.10
cuda 12.1
torch==2.3.1
torchvision==0.18.1
```

### PR DESCRIPTION
**Pull request by ispobock**
## Motivation

Fix https://github.com/sgl-project/sglang/issues/735.
It seems an issue of outlines. The `pad_token_id` in [chatglm3-6b](https://huggingface.co/THUDM/chatglm3-6b) is 0 and immutable, but outlines tries to reset it. 

## Modification
Make the `pad_token_id` property mutable and reset it back.

### CODE REVIEW

**Comment by zhyncs** [comment]
ref https://github.com/sgl-project/sglang/issues/735

File: python/sglang/srt/constrained/fsm_cache.py [hunk 1]
```diff
@@ -21,7 +21,27 @@ def __init__(self, tokenizer_path, tokenizer_args_dict, enable=True):
             tokenizer = AutoTokenizer.from_pretrained(
                 tokenizer_path, **tokenizer_args_dict
             )
-            self.outlines_tokenizer = TransformerTokenizer(tokenizer)
+            try:
+                self.outlines_tokenizer = TransformerTokenizer(tokenizer)
+            except AttributeError:
```
Comment (zhyncs) [inline python/sglang/srt/constrained/fsm_cache.py:26]: When the error is not about `pad_token_id`, we may still raise agian.

**Comment by zhyncs** [comment]
ref https://github.com/outlines-dev/outlines/pull/1068

**Review by zhyncs — APPROVED** [review]

File: python/sglang/srt/constrained/fsm_cache.py (hunk 1 above)
Comment (ispobock) [reply python/sglang/srt/constrained/fsm_cache.py:26]: The `AttributeError` seems only happen for `pad_token_id` in the TransformerTokenizer initialization.

**Pull request merged**